hris_publisher/publish.py

```diff
@@ -0,0 +1,145 @@
+        vault_record = v.find_by_email(email)
+
+        # Enrich the profile with the new data fields from HRIS extract. (Groups only for now)
+        if vault_record is not None:
+            logger.info('User record located in identity vault for: {user}'.format(user=email))
+            current_group_list = vault_record.get('groups')
+
+            for group in current_group_list:
+                # Remove existing groups with hris_ attrs from assertions.
+                if group.find('hris_') == 0:
+                    logger.info('Removing current hris_ assertion from {user} for group : group'.format(
+                            user=email,
+                            group=group
+                        )
+                    )
+                    current_group_list.pop(group)
+
+            # Add the new assumptions we made about the user to the current grouplist
+            vault_record['groups'] = current_group_list + hris_groups
+            logger.info('Groups reintegrated for user: {user}'.format(user=email))
+            # Add a signature XXX TBD
+
+            # Send to CIS Validator Function
+            logger.info('Sending profile to CIS for: {user}'.format(user=email))
+
+            sts = boto3.client('sts')
+            sts_response = sts.assume_role(
+                RoleArn=config('iam_role_arn', namespace='cis'),
+                RoleSessionName=config('iam_role_session_name', namespace='cis')
+            )
+            logger.info('CIS Publisher role assumed.')
+
+            cis_publisher_session = boto3.session.Session(
+                aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
+                aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
+                aws_session_token=sts_response['Credentials']['SessionToken'],
+                region_name='us-west-2'
+            )
+
+            publisher = {
+                'id': config('publisher_name', namespace='cis')
+            }
+
+            data = {
+                'user_id': vault_record.get('user_id'),
+                'timezone':  vault_record.get('timezone'),
+                'active': vault_record.get('active'),
+                'lastModified': vault_record.get('lastModified'),
+                'created': vault_record.get('created'),
+                'userName': vault_record.get('userName'),
+                'displayName': vault_record.get('displayName'),
+                'primaryEmail': vault_record.get('primaryEmail'),
+                'emails': vault_record.get('emails'),
+                'uris': vault_record.get('uris'),
+                'picture': vault_record.get('picture'),
+                'shirtSize': vault_record.get('shirtSize'),
+                'groups':vault_record.get('groups'),
+                'firstName': vault_record.get('firstName'),
+                'lastName': vault_record.get('lastName'),
+
+                # Hardcoded fields these can not currently be set in profile editor.
```

Comment:
Would add "XXX" so we can grep for that as it'll fail validation if the profile contains different data at some point

Reply:
Yes can do.